web-extension/search.js

```diff
@@ -121,26 +127,39 @@ function _onSearchResults(response, isHostQuery) {
     searching = false;
 }
 
-function _onResultSelected(event, element) {
-    element = element || event.target;
-    const value = element.innerText;
+function _onEntryAction(event, element) {
+    if (element) {
+        event.target = element;
+    }
+
     if (event.altKey) {
-        sendNativeAppMessage({ type: 'getData', entry: value }).then(
-            message => onEntryData(element, message),
-            logAndDisplayError
-        );
+        _onEntryDetails(event);
     } else if (event.shiftKey) {
-        sendNativeAppMessage({ type: 'getLogin', entry: value }).then(
-            _onLoginCredentialsDoCopyClipboard,
-            logAndDisplayError
-        );
+        _onEntryCopy(event);
     } else {
+        const value = element.innerText;
         browser.runtime
             .sendMessage({ type: 'LOGIN_TAB', entry: value, tab: { id: currentTab.id, url: currentTab.url } })
             .then(_onLoginCredentialsDidLogin, logAndDisplayError);
     }
 }
 
+function _onEntryCopy(event) {
+    const value = event.target.innerText;
+    sendNativeAppMessage({ type: 'getLogin', entry: value }).then(
+        _onLoginCredentialsDoCopyClipboard,
+        logAndDisplayError
+    );
+}
+
+function _onEntryDetails(event) {
```

Comment:
_onEntryCopy and _onEntryDetails don't need the full event, maybe just pass the "target" element explicitly?